agent/agentscripts/agentscripts_other.go

```diff
@@ -0,0 +1,20 @@
+//go:build !windows
+
+package agentscripts
+
+import (
+	"os/exec"
+	"syscall"
+)
+
+func cmdSysProcAttr() *syscall.SysProcAttr {
+	return &syscall.SysProcAttr{
+		Setsid: true,
+	}
+}
+
+func cmdCancel(cmd *exec.Cmd) func() error {
+	return func() error {
+		return syscall.Kill(-cmd.Process.Pid, syscall.SIGINT)
```

Comment:
When e.g. `bash` is executing a script, `SIGINT` gets captured and passed to the current command being executed, this results in that command exiting and then the script executes the _next_ command. Essentially circumventing what we're trying to do here.

Changing this to `SIGHUP` will work better for what we're trying to do and will ensure everything is canceled. This is what a terminal typically sends to a program when the attached terminal goes away, so seems alright.

The downside is that perhaps it's more common to have handling for `SIGINT`, in which case this could cause a dirty exit for some script authors. But we can just document this behavior instead so it's a non-issue and can still be handled.

Reply:
That makes sense. I appreciate the thorough thought here! 😍